### ISSUE TITLE
Clang allows global namespace qualification before template specialization

### ISSUE THREAD

**Issue by kim366** [llvm/llvm-project#107582]
The following C++ code compiles successfully in Clang (trunk):

```cpp
template<typename>
struct X;

template<>
struct ::X<int> {};
```

I am unsure what the C++ standard has to say about this, but gcc fails with the error `global qualification of class name is invalid before '{' token`. If this is incorrect usage, there should at least be a compiler warning.

**Comment by shafik** [comment]
Based on [CWG355](https://www.open-std.org/jtc1/sc22/wg21/docs/cwg_defects.html#355) this looks like a gcc bug, see: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66892

gcc is the only one to reject: https://godbolt.org/z/96TYzYe1j

although clang trunk does generate a diagnostic:

```console
<source>:5:10: warning: extra qualification on member 'X' [-Wextra-qualification]
    5 | struct ::X<int> {};
      |          ^
```

**Comment by kim366** [comment]
Thanks for the quick reply. Then all is good and I will close the issue.
